src/k8s-engine/lima.ts

```diff
@@ -1812,6 +1815,40 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     await this.ssh('sudo', 'update-ca-certificates');
   }
 
+  protected async getHostIPAddr(): Promise<string> {
+    try {
+      const lines = (await this.limaWithCapture('shell', '--workdir=.', MACHINE_NAME, 'ip', 'route', 'list', 'eth0')).split(/\n/);
+      const fields = lines[0].split(/\s+/);
+
+      return fields[2];
+    } catch (err: any) {
+      console.log(`ip route failed: ${ err }`, err);
+      throw err;
+    }
+  }
+
+  protected async installCredentialHelper() {
+    const credsPath = getServerCredentialsPath();
+
+    try {
+      const hostIPAddr = await this.getHostIPAddr();
+      const stateInfo: ServerState = JSON.parse(await fs.promises.readFile(credsPath, { encoding: 'utf-8' }));
+      const escapedPassword = stateInfo.password.replace(/\\/g, '\\\\')
+        .replace(/"/g, '\\"');
+      const fileContents = `CREDFWD_AUTH="${ stateInfo.user }:${ escapedPassword }"
+CREDFWD_URL="http://${ hostIPAddr }:${ stateInfo.port }"
```

Comment:
Any reason not to use single quotes here instead of double quotes? All the variable replacement is done in the TypeScript, so that written-out file shouldn't have any. (Otherwise we'll need to escape dollar too.)

Reply:
True. Still have the issue of escaping single-quotes (I realize right now passwords are pure alphanumeric strings, but I want to allow for the day users can set passwords via environment variables and I don't want to expect that the dev who turns that on will look here to see if there are problematic characters).